base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -541,6 +546,12 @@ public void onCapabilityRetrieved(Object capability) {
                         int currentWindowID = windowCapability.getWindowID() != null ? windowCapability.getWindowID() : PredefinedWindows.DEFAULT_WINDOW.getValue();
                         if (currentWindowID == PredefinedWindows.DEFAULT_WINDOW.getValue()) {
                             defaultMainWindowCapability = windowCapability;
+
+                            KeyboardCapabilities keyboardCapabilities = windowCapability.getKeyboardCapabilities();
+                            if (keyboardCapabilities.getMaskInputCharactersSupported() != null
```

Comment:
@kostyaBoss On line 551 you are checking if `getMaskInputCharactersSupported () != null` .  However, that method returns a `Boolean`.  Do you need to change this logic?

Reply:
I don't think so, because according to the proposal this is not a mandatory field, as far as I can remember, this means that calling `getMaskInputCharactersSupported` may produce NPE. That was the reason for this additional check